### PR TITLE
#170947690 rate accomodations

### PR DESCRIPTION
**Pull request by Kagorora**
#### What does this PR do?

- Allow users to rate accommodations

#### Description of Task to be completed?

- Have the following endPoint work:

   GET     /edit/accommodation/rate/:id

#### How should this be manually tested?

- **login as any user**

       { 
	     "email": "alain.maxime@gmail.com", 
	     "password": "Niyonkuru@1" 
       }

- **view all the accommodation to get the id**

       GET    /api/v1/view/accommodations

-**Rate the accommodation**

       PATCH     /api/v1/edit/accommodation/rate/:id

                        eg: /api/v1/edit/accommodation/rate/1

- **View all to see the average rate**

       GET     http://localhost:4000/api/v1/view/accommodations

#### Any background context you want to provide?

- Only existing accommodations can be rated

#### What are the relevant pivotal tracker stories?

- [#170947690](https://www.pivotaltracker.com/story/show/170947690)

#### Screenshots (if appropriate)

-  **login as any user**

Screen Shot 2020-03-12 at 10 42 55

- **view all the accommodation**

Screen Shot 2020-03-12 at 10 41 29

-**Add on your rate(ADD RATE IF IT IS THE FIRST TIME RATING THAT ACCOMMODATION )**

Screen Shot 2020-03-11 at 17 11 18

-**Update your rate(IF YOU HAVE RATED THAT ACCOMMODATION BEFORE)**

Screen Shot 2020-03-12 at 10 41 06

-**To view your impact of your rate on the accommodation**

Screen Shot 2020-03-12 at 10 41 29

#### Questions:

-N/A

### CODE REVIEW

**Review by Moise1 — APPROVED** [review]

File: src/db/migrations/20200213181730-create-user.js [hunk 1]
```diff
@@ -6,44 +6,44 @@ module.exports = {
       allowNull: false,
       autoIncrement: true,
       type: Sequelize.INTEGER,
-      primaryKey: true,
+      primaryKey: true
```
Comment (kagabof) [inline src/db/migrations/20200213181730-create-user.js]: why did you edit this file?

File: .codeclimate.yml [hunk 2]
```diff
@@ -33,6 +33,8 @@ checks:
 exclude_patterns:
 - "src/config/"
 - "src/db/"
+- "src/helpers/commentQueries.js"
+- "src/helpers/accomodationQueries.js"
```
Comment (kagabof) [inline .codeclimate.yml]: why did you exclude those files?

File: .eslintrc.js [hunk 3]
```diff
@@ -7,7 +7,6 @@ module.exports = {
   root: true,
   'extends': 'airbnb-base',
   env: {
-    es6: true,
     node: true,
```
Comment (kagabof) [inline .eslintrc.js]: Why did you remove es6?

File: src/controllers/rate.accommodation.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
+import models from '../db/models';
```
[the remark this answers is not in the document]
Comment (kagabof) [reply src/controllers/rate.accommodation.js:1]: Fix this hound issue

**Review by kagabof — CHANGES_REQUESTED** [review]
Please resolve all those hound issues.

File: src/db/migrations/20200213181730-create-user.js (hunk 1 above)
Comment (Kagorora) [reply src/db/migrations/20200213181730-create-user.js]: That was some eslint errors, to fix them I added a trailing comma

File: .codeclimate.yml (hunk 2 above)
Comment (Kagorora) [reply .codeclimate.yml]: The reason is, They very look alike so codeclimate was claiming the similarity however they are not performing the same task at all and I need both those files as they help in not repeating queries(findOne, findAll, ...)

**Review by T2Wil — APPROVED** [review]
Nice work @Kagorora .

File: src/controllers/rate.accommodation.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (Kagorora) [reply src/controllers/rate.accommodation.js:1]: fixed

File: .eslintrc.js (hunk 3 above)
Comment (Kagorora) [reply .eslintrc.js]: I removed it by mistake , I have added it back

File: .codeclimate.yml (hunk 2 above)
Comment (kagabof) [reply .codeclimate.yml]: So you should fix it then.

File: .eslintrc.js (hunk 3 above)
Comment (kagabof) [reply .eslintrc.js]: Why did you add the browser?

File: src/db/migrations/20200220144524-create-request.js [hunk 5]
```diff
@@ -6,7 +6,7 @@ module.exports = {
       autoIncrement: true,
       primaryKey: true,
       type: Sequelize.INTEGER,
-      unique: true,
+      unique: true
```
Comment (kagabof) [inline src/db/migrations/20200220144524-create-request.js]: Why are you editing this file?

File: src/db/migrations/20200304100142-create-comment.js [hunk 6]
```diff
@@ -13,8 +13,8 @@ module.exports = {
       references: {
         model: 'Requests',
         key: 'id',
-        as: 'id',
-      },
+        as: 'id'
```
Comment (kagabof) [inline src/db/migrations/20200304100142-create-comment.js]: why did you edit this file?

**Review by kagabof — CHANGES_REQUESTED** [review]
Please don't edit files that you are not supposed to edit.

File: src/db/migrations/20200220144524-create-request.js (hunk 5 above)
Comment (Kagorora) [reply src/db/migrations/20200220144524-create-request.js]: Those are eslint errors that require a trailing comma

File: src/db/migrations/20200304100142-create-comment.js (hunk 6 above)
Comment (Kagorora) [reply src/db/migrations/20200304100142-create-comment.js]: Same as the other ones, Once I open the file there are requesting to fix eslint lint

**Review by kagabof — APPROVED** [review]
